examples/cosmwasm/zero-to-hero/vote.qnt

```diff
@@ -0,0 +1,284 @@
+/// A Quint specification following Callum-A's cosmwasm-zero-to-hero tutorial:
+/// https://github.com/Callum-A/cosmwasm-zero-to-hero
+///
+/// Numbers in square brackets [nn] refer to the relevant chapter.
+
+module vote {
+
+  /*** TYPES ***/
+
+  // 05 - Contract state
+  // https://github.com/Callum-A/cosmwasm-zero-to-hero/tree/main/05%20-%20State
+
+  type Addr = str
+  val None : Addr = ""
+
+  type UUID = str
+
+  type Config = {
+    admin: Addr,
+  }
+
+  type Poll = {
+    creator: Addr,
+    question: str,
+    options: List[(str, int)]
```

Comment:
Ideally, this would be a `str -> int` map, but apparently CosmWasm storage has no first-class support for maps.

I don't trust the logic around this, so using the original encoding here may be a good showcase for Quint :)

Reply:
I guess, they are avoiding maps, since maps normally do not have deterministic iteration

Reply:
introduce `type u64 = int` and say in a comment that we do not account for overflows/underflows yet?

Reply:
> I guess, they are avoiding maps, since maps normally do not have deterministic iteration

Tutorial says: "Some users of other programming languages may suggest using a Map structure however these cannot be stored in cosmwasm storage plus."

Logically, it should be a map, but I'm not sure their logic really implements a map.